checkmk_extensions/agent/robotmk_collector.ps1

```diff
@@ -1,21 +1,30 @@
 Set-StrictMode -Version 3.0
 $ErrorActionPreference = "Stop"
 
+function Main {
+    Param (
+        [parameter(Mandatory=$true, Position=0)]
+        [string]$ConfigPath
+    )
+
+    $resultsDirectory = GetResultsDirectory($ConfigPath)
+    $files = Get-ChildItem -File -Recurse $resultsDirectory
+
+    Write-Output "<<<robotmk_v2:sep(10)>>>"
+    foreach ($file in $files) {
+        Write-Output (Get-Content -Raw -ErrorAction 'SilentlyContinue' $file.FullName)
+    }
+}
+
 function GetResultsDirectory {
     [OutputType([string])]
     Param (
         [parameter(Mandatory=$true, Position=0)]
-        [string]$configPath
+        [string]$ConfigPath
     )
-    $configFileContent = Get-Content -Raw -Path $configPath
+    $configFileContent = Get-Content -Raw -Path $ConfigPath
     $configData = ConvertFrom-Json -InputObject $configFileContent
     return $configData.results_directory -as [string]
 }
 
-$resultsDirectory = GetResultsDirectory("${env:MK_CONFDIR}\robotmk.json")
-$files = Get-ChildItem -File -Recurse $resultsDirectory
-
-"<<<robotmk_v2:sep(10)>>>"
-foreach ($file in $files) {
-	Get-Content -ErrorAction 'SilentlyContinue' $file.FullName
-}
+Main("${env:MK_CONFDIR}\robotmk.json")
```

Comment:
OTI: This is an unconventional signature for a main function.

Reply:
Yes, that's true. It would definitely be more standard to pass command line arguments (such as a path to a config file) to `Main`, but agent plugins are always called without any arguments. Would you prefer having `"${env:MK_CONFDIR}\robotmk.json"` inside `Main`?

Reply:
I think it might be easier to test this way, so no =)